src/Mollie/WC/Plugin.php

```diff
@@ -276,6 +284,29 @@ function ($settings) {
 		self::$initiated = true;
 	}
 
+    public static function maybeLetWCCancelOrder($willCancel, $order) {
+        if (!empty($willCancel)) {
+            if ($order->get_payment_method()
+                !== 'mollie_wc_gateway_banktransfer'
+            ) {
+                return $willCancel;
+            }
+            //is banktransfer due date setting activated
+            $banktransferSettings = get_option(
+                'mollie_wc_gateway_banktransfer_settings'
+            );
+            $expiry_days
+                = $banktransferSettings['activate_expiry_days_setting'];
+
+            $dueDateActive = mollieWooCommerceStringToBoolOption(
+                $expiry_days
+            );
```

Comment:
I guess it would be a good idea to have a separate method for this. If I understand everything correctly, it can be reused [here](https://github.com/mollie/WooCommerce/blob/MOL-182/src/Mollie/WC/Plugin.php#L737-L742).